### PR TITLE
MPDP 219 Search Results Page

### PR DESCRIPTION
**Pull request by SidG-CG**

### CODE REVIEW

File: app/routes/search/index.ts [hunk 1]
```diff
@@ -1,34 +1,102 @@
...
+  const previous = requestedPage <= 1 ? null : {
+    href: `${path}?searchString=${encodedSearchString}&page=${requestedPage - 1}`,
+    labelText: `${requestedPage - 1} of ${totalPages} `,
+    attributes: {
+      id: 'prevOption'
+    }
+  }
+  
+  const next = totalPages <= 1 || totalPages === requestedPage ? null : {
+    href: `${path}?searchString=${encodedSearchString}&page=${requestedPage + 1}`,
```
Comment (vmuthu-kn) [inline app/routes/search/index.ts:19]: is this not supposed to be requestedPage + limit

File: app/routes/search/index.ts [hunk 2]
```diff
@@ -1,34 +1,102 @@
...
+        if(!request.query.searchString) {
+          return h.view('search/index')
+        }
+
+        return h.view('search/results', createModel(request.query))
+      }
+    }
+  },
+  {
+    path: '/search',
```
Comment (vmuthu-kn) [inline app/routes/search/index.ts:83]: why do we need 2 search api paths with GET and POST. we are only reading data why POST required?

File: app/views/macros/search-results.njk [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+            </a>
+          </h3>
+          <div>
+            <div class="govuk-heading-s search-results-amount">{{ item.amount }}</div>
+          </div>
+        </div>
+        <div class="govuk-body subtitle margin-bottom-20 govuk-!-font-size-16">Location: {{ item.part_postcode }}, {{item.town}}, {{item.county_council}}</div>
+        <hr class="govubk-section-break govuk-section-reak--visible section-break-heading--lightgray">
+      </div>
+    {% else %}
```
Comment (vmuthu-kn) [inline app/views/macros/search-results.njk:20]: are we combining if else block from different file? is it good for readability?

Comment (SidG-CG) [reply app/views/macros/search-results.njk:20]: This is a nunjucks thing called for else block. Very bizarre, I know but its supported by the engine, https://mozilla.github.io/nunjucks/templating.html#for

File: app/routes/search/index.ts (hunk 2 above)
Comment (SidG-CG) [reply app/routes/search/index.ts:83]: So having looked at this we do need both endpoint.
The GET endpoint is required for 

- when you come to the search page for the first time
- when you click on next on pagination. This contains links with query string and get request uses query object to appropriately send results for the requested page.

The POST endpoint is required for:
- When a user searches though the search input box on the search page and on the results page. As this uses form, the request receives payload as opposed to the query object on the GET endpoint.

Also both endpoint behave differently when no searchString is present. The GET endpoint responds back with the search index page. The POST endpoint responds back with an error on the index page.

So overall we do need both endpoints as they are slightly doing different things and responding to different kind of request.

Comment (vmuthu-kn) [reply app/routes/search/index.ts:83]: Even though it is using the differently(querystring/payload) both are doing the read operation from the server. Post should only used for write operation for example creating the payment data in this case as per rest standard. to change the behaviour we can return diff result based on the bask-end condition.

Comment (vmuthu-kn) [reply app/routes/search/index.ts:83]: as dicussed with Sid , this is approved in MPDP-228

**Review by vmuthu-kn — APPROVED** [review]

**Pull request merged**